src/main/java/luke/Luke.java

```diff
@@ -0,0 +1,301 @@
+        String[] inputArr = userInput.split(" ");
+        String command = inputArr[0];
+
+        if (command.equalsIgnoreCase("bye")) {
+            executeCommand(CommandType.BYE, inputArr);
+        } else if (command.equalsIgnoreCase("list")) {
+            executeCommand(CommandType.LIST, inputArr);
+        } else if (command.equalsIgnoreCase("mark")) {
+            try {
+                executeCommand(CommandType.MARK, inputArr);
+            } catch (LukeException e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("unmark")) {
+            try {
+                executeCommand(CommandType.UNMARK, inputArr);
+            } catch (LukeException e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("todo")){
+            try {
+                executeCommand(CommandType.TODO, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("deadline")){
+            try {
+                executeCommand(CommandType.DEADLINE, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("event")) {
+            try {
+                executeCommand(CommandType.EVENT, inputArr);
+            } catch (InsufficientArguments e) {
+                printReply(e.getMessage());
+            }
+        } else if (command.equalsIgnoreCase("delete")) {
+            try {
+                executeCommand(CommandType.DELETE, inputArr);
+            } catch (InsufficientArguments | IncorrectInput e) {
+                printReply(e.getMessage());
+            }
+        } else {
+            throw new InvalidCommand("Invalid command");
+        }
+    }
+
+    private static void printGreeting() {
+        printDivider();
+        System.out.println(LOGO);
+        System.out.println("Hi im Luke!\nWhat can I do for you? :)");
+        printDivider();
+    }
+
+    private static void loadSingleTask(String taskStr) {
+        String[] taskStrArr = taskStr.split("\\|");
+        switch (taskStrArr[TASK_TYPE_INDEX]) {
+        case "T" -> tasks.add(new ToDo(taskStrArr[DESCRIPTION_INDEX], taskStrArr[ISDONE_INDEX].equals("1")));
+        case "D" -> tasks.add(new Deadline(taskStrArr[DESCRIPTION_INDEX], taskStrArr[BY_INDEX],
+                taskStrArr[ISDONE_INDEX].equals("1")));
+        case "E" -> tasks.add(new Event(taskStrArr[DESCRIPTION_INDEX], taskStrArr[FROM_INDEX], taskStrArr[TO_INDEX],
+                taskStrArr[ISDONE_INDEX].equals("1")));
+        }
+    }
+
+    private static void loadSavedTasks() throws IOException {
+        File dataDir = new File("data");
+        if (!dataDir.exists()) {
+            dataDir.mkdir();
+        }
+        File save = new File("data/tasks.txt");
```

Comment:
There are still quite a number of magic strings lying around the codebase. Please take a look!